extensions/service_bindings_manager.go

```diff
@@ -0,0 +1,76 @@
+package extensions
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// ServiceBinding is a representation of a service binding that has
+// been provided into the build container.
+type ServiceBinding struct {
+	// Name is the name of the directory that binding was contained within.
+	Name string
+
+	// Type is the content of the type file contained within the binding
+	// directory.
+	Type string
+
+	// Provider is the content of the provider file contained within the binding
+	// directory.
+	Provider string
+
+	// Path is the filepath of the binding directory.
+	Path string
+
+	// Secrets is a mapping of file name to contents for all other files
+	// contained in the bindings directory.
+	Secrets map[string][]byte
+}
+
+// ListServiceBindings returns all service bindings found in the given root
+// directory. These bindings must conform the the Kubernetes Service Binding
+// specification as documented in https://github.com/k8s-service-bindings/spec.
+func ListServiceBindings(root string) ([]ServiceBinding, error) {
```

Comment:
Forgive me if this isn't the pattern we've followed for packit, we should consider placing this function into an interface for mockability. Then we can test consumers (e.g. when we add [service bindings to the nginx buildpack](https://github.com/paketo-buildpacks/packit/issues/107)) without needing to write files to the filesystem.

Reply:
I'm on board with this, given bindings' directory structures can be [pretty elaborate](https://github.com/buildpacks/spec/blob/main/extensions/bindings.md#example-directory-structure). It would be nice to be able to test buildpacks that consume bindings without having to write these structures to the filesystem.